network changes for new localhost api, still needs fixing

mattbarker016 opened the pull request:
I added the new Network base URL. I also removed the old BusStop functions throughout the project. 

Currently, while listening on 3000, the `parseAllStops` function receives a JSON that it isn't expecting. The search bar will crash on tapping because the proper data wasn't received from the API call / put into the JSON. I changed `json.array!` to `json.arrayValue` to allow this for now, but this is where things go screwy.

**Network.swift, line 22-24**
````    
required init(json: JSON) throws {
    allStops = parseAllStops(json: json.array!)
}
````


<img width="922" alt="screen shot 2017-08-28 at 3 09 40 pm" src="https://user-images.githubusercontent.com/8039928/29789092-f0e517cc-8c02-11e7-9341-9d30a9f0d8ba.png">

mattbarker016 committed:
network changes for new localhost api, still needs fixing

made json array optional to not crash

fixed all bus stops

fixed warning

added a network function called getLocationFromPlaceId

on the fly network changes

mattbarker016 committed:
network changes for new localhost api, still needs fixing

mattbarker016 committed:
made json array optional to not crash

AAAstorga committed:
fixed all bus stops

AAAstorga committed:
fixed warning

AAAstorga committed:
added a network function called getLocationFromPlaceId

dantheli commented on TCAT/Utilities/Network.swift:
```diff
@@ -9,27 +9,30 @@ import Foundation
+            let long = location["latitude"].doubleValue
             let busStop = BusStop(name: name, lat: lat, long: long)
             allStopsArray.append(busStop)
         }
```
Could be nicer in a map

dantheli reviewed (approved):
Looks good. Was there a specific reason why you updated the platform in podfile to 9.3?

mattbarker016 committed:
on the fly network changes

mattbarker016 committed:
network fix, options vc cleanup and error alert for no location

AAAstorga committed:
added support for place results

network fix, options vc cleanup and error alert for no location

prep for debugging routeDetail

route json fixes for backend changes

route fixes

tentative fix for bus icon rount number alignment @monica

AAAstorga committed:
added support for place results

AAAstorga committed:
updates

mattbarker016 committed:
prep for debugging routeDetail

mattbarker016 committed:
route json fixes for backend changes

mattbarker016 committed:
route fixes

mattbarker016 committed:
tentative fix for bus icon rount number alignment @monica

myo3 committed:
Updated OptionVC to use new Network.getRoute()

myo3 committed:
Updated OptionVC to use new Network.getRoute()

Fixed logic glitch w/ searchForRoutes()

it moves sweet

fixed no network bugs

myo3 committed:
Fixed logic glitch w/ searchForRoutes()

myo3 committed:
Removed Pod folder. Updated gitignore so no longer track Pod folder

myo3 committed:
Sublassed BusStop and PlaceResult from Place. Removed Podfile.lock

AAAstorga committed:
fixed no network bugs

AAAstorga committed:
Merge branch 'matt/route_detail' of https://github.com/cuappdev/tcat-ios into matt/route_detail

mattbarker016 committed:
it moves sweet

mattbarker016 committed:
implemented live tracking, refactored Direction model

mattbarker016 committed:
changes around app for new Direction model

mattbarker016 committed:
changes around app for new Direction model

myo3 committed:
Connect RouteSummaryObject to Route object and delete mainStopNum & mainStops (in progress)

myo3 committed:
Refactored RouteSummaryObject in OptionsVC (but not in RouteDetailVC)

myo3 committed:
Fix merge conflicts (LargeDetailTableViewCell, Directions, & RouteDetailVC will crash)

myo3 committed:
Fixed errors in RouteDetailVC

myo3 committed:
Delete Podfile.lock from Github
